chainerx_cc/chainerx/routines/arithmetic.cc

```diff
@@ -491,4 +491,34 @@ Array Power(const Array& x1, Scalar x2) { return internal::Binary(&PowerASImpl,
 
 Array Power(Scalar x1, const Array& x2) { return internal::Binary(&PowerSAImpl, x1, x2, GetArithmeticResultDtype(x1, x2)); }
 
+Array Fmod(const Array& x1, const Array& x2) {
+    CheckEqual(x1.shape(), x2.shape());
+    Array out = Empty(x1.shape(), x1.dtype(), x1.device());
+
+    {
+        NoBackpropModeScope scope{};
+        x1.device().backend().CallKernel<FmodKernel>(x1, x2, out);
+    }
+
+    {
+        BackwardBuilder bb{"fmod", x1, x2};
+        if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+            bt.Define([dtype = x1.dtype()](BackwardContext& bctx) {
+                const Array& gx = *bctx.output_grad();
+                bctx.input_grad() = dtype == gx.dtype() ? std::move(gx) : gx.AsType(dtype);
```

Comment:
@aksub99 , I don't think it is a good idea to move from `gx` i.e. `*bctx.output_grad()` as it is utilised for lambda below and that may lead to undefined behaviour and segfaults. That should probably fix the crash in CI as well.

Reply:
Seems in other places which uses `std::move` declares source variable non const qualified that isn't reference.
C++ is hard...

Reply:
Thanks a lot @kshitij12345 and @take-cheeze.I have made the suggested changes.